### PR TITLE
SP2-887 input validation

### PR DESCRIPTION
**Pull request by alexw-moj**
Adds input validation for text fields and autocomplete boxes

### CODE REVIEW

File: integration_tests/e2e/achieve-goal.cy.ts [hunk 1]
```diff
@@ -71,6 +72,20 @@ describe('Achieve goal', () => {
       cy.get('.govuk-details__text').should('contain', 'Some optional text in the achievement note field')
     })
 
+    it('Confirm errors are displayed with optional note of more than 4000 characters', () => {
```
Comment (pipwilson) [inline integration_tests/e2e/achieve-goal.cy.ts:75]: For this (and the other tests which exercise the error message on length of field) can we also validate that the relevant field contains the repopulated value please?

File: server/routes/update-goal/models/UpdateGoalPostModel.ts [hunk 2]
```diff
@@ -15,6 +15,10 @@ export class StepModel {
 }
 
 export default class UpdateGoalPostModel {
+  @Expose()
```
Comment (pipwilson) [inline server/routes/update-goal/models/UpdateGoalPostModel.ts]: Do we need Expose() here? Shouldn't it be exposed by default?

File: integration_tests/e2e/achieve-goal.cy.ts (hunk 1 above)
Comment (alexw-moj) [reply integration_tests/e2e/achieve-goal.cy.ts:75]: Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/199/commits/427fbb28b69b11c9d45de2c96503d8008ebaeb50

File: server/routes/update-goal/models/UpdateGoalPostModel.ts (hunk 2 above)
Comment (alexw-moj) [reply server/routes/update-goal/models/UpdateGoalPostModel.ts]: Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan-ui/pull/199/commits/427fbb28b69b11c9d45de2c96503d8008ebaeb50

**Review by pipwilson — APPROVED** [review]
LGTM

**Pull request merged**